analyses/focal-cn-file-preparation/01-add-ploidy-cnvkit.Rmd

```diff
@@ -17,19 +17,42 @@ library(dplyr)
 ### Read in data
 
 ```{r}
-cnvkit_file <- file.path("..", "..", "data", "cnv-cnvkit.seg.gz")
+cnvkit_file <- file.path("..", "..", "data", "v8", "cnv-cnvkit.seg.gz")
 cnvkit_df <- readr::read_tsv(cnvkit_file)
 ```
 
 ```{r}
-histologies_file <- file.path("..", "..", "data", "histologies.tsv")
+histologies_file <- file.path("..", "..", "data", "v8", "histologies.tsv")
 histologies_df <- readr::read_tsv(histologies_file, guess_max = 100000)
 ```
 
+
+# Edits for https://github.com/PediatricOpenTargets/ticket-tracker/issues/177
+# For PBTA, histologies file has germline_sex_estimate already assigned to WXS patients which have had WGS performed
+# Updating here germline_sex_estimate for cohort=="TARGET" where is.na(germline_sex_estimate)
+#Isolate IDs where germline_sex_estimate is NA and cohort == "TARGET"
+germline_sex_estimate_na <- histologies_df %>% filter(cohort == "TARGET") %>% filter(is.na(germline_sex_estimate))
+na_ids <- germline_sex_estimate_na %>% pull(Kids_First_Biospecimen_ID)
+
+#Split the dataset into two groups, fix data for select IDs
+histology_subset1 <- histologies_df %>% filter(!Kids_First_Biospecimen_ID %in% na_ids)
+
+histology_subset2 <- germline_sex_estimate_na %>% mutate(germline_sex_estimate = reported_gender)
```

Comment:
```suggestion
germline_sex_estimate_na <- histologies_df %>% 
filter(cohort == "TARGET" & is.na(germline_sex_estimate))

na_ids <- germline_sex_estimate_na %>% 
pull(Kids_First_Biospecimen_ID)

#Split the dataset into two groups, fix data for select IDs
histology_subset1 <- histologies_df %>% 
filter(!Kids_First_Biospecimen_ID %in% na_ids)

histology_subset2 <- germline_sex_estimate_na %>% 
mutate(germline_sex_estimate = reported_gender)
```

hi @sangeetashukla (and cc @runjin326 because I noticed this on some of your PRs as well) - I suggest we do not use a long row format for commands, but split these by row for readability  - this also helps with reviewing lines. So, I wouldn't commit this from the web, but rather, update the code in R so you can get the styling/indentation correct. I also got rid of one of your redundant filters.

I also have a suggestion for this in general - I think we should update this without regard for cohort. For instance, we may get other cohorts of WXS (in fact, we will) and instead of adding cohorts iteratively, I suggest we have some different logic here. The logic would be if `experimental_strategy == WXS & is.na(germline_sex_estimate)`, then does WGS for this same sample exist (via `sample_id`) and does WGS `germline_sex_estimate` exist? if yes, mutate to use WGS `germline_sex_estimate`. If no, mutate to use `reported_gender.`

Reply:
@jharenza, thanks for the comment! I will break my lines from now on!
And to you comment, yes, it makes sense to check WGS - but as I mentioned in my previous comment & confirmed by Krutika, looks like PBTA already have WGS fixed. 
And @sangeetashukla and I also checked that for TARGET, no samples with `germline_sex_estimate == NA` has matching WGS. I just did a full check for all cohorts - looks like none of them have matching WGS in PBTA and TRAGET? But maybe we can pull from GMKF? 
```
> na_sex_participant <- histology %>% filter(experimental_strategy == "WXS") %>% filter(sample_type == "Tumor") %>% filter(is.na(germline_sex_estimate))  %>% pull(Kids_First_Participant_ID)
> histology %>% filter(Kids_First_Participant_ID %in% na_sex_participant) %>% select(cohort, experimental_strategy) %>% table()
        experimental_strategy
cohort   RNA-Seq  WGS  WXS
  GMKF         8   20    0
  PBTA        18    0   32
  TARGET     531    0 1286
```
We can definitely add the check for WGS in the workflow and change to gender when WGS does not exist still.

Reply:
>And to you comment, yes, it makes sense to check WGS - but as I mentioned in my previous comment & confirmed by Krutika, looks like PBTA already have WGS fixed.

Oh yes, PBTA is currently OK, but in v10, we will have some additional WXS from PBTA which will not have WGS, so would be good to implement this logic now.

Reply:
@jharenza, good point! Let's implemented the WGS logic then! (cc @sangeetashukla)

Reply:
Done